lib/Telemetry/src/Dispatcher.h

```diff
@@ -0,0 +1,60 @@
+#include "Particle.h"
+#include "DataQueue.h"
+#include "IntervalLogger.h"
+
+#ifndef _DISPATCHER_H_
+#define _DISPATCHER_H_
+
+/**
+ * Dispatcher owns and operates on a collection of loggers
+ **/
+class Dispatcher {
+    public:
+        /**
+         * Constructs a Dispatcher with loggers, numLoggers and DataQueue
+         **/
+        Dispatcher(IntervalLogger **loggers, uint16_t numLoggers, DataQueue *dataQ) {
+            _loggers = loggers;
+            _numLoggers = numLoggers;
+            _dataQ = dataQ;
+            _logThisLoop = false;
+        }
+
+        /**
+         *  Must be called from main loop!  Takes times since program start (in seconds) and checks whether
+         *  log needs to be called on any of its loggers.
+         * 
+         * @param time the time in seconds since the start of the program
+         **/
+        void run(unsigned long time) {
+            // check if it's time to log any data from any of the loggers
+            for (uint16_t i = 0; i < _numLoggers; i++) {
+                if ((_loggers[i]->getLastLog() + _loggers[i]->getInterval()) <= time) {
+                    _loggers[i]->setLastLog(time);
```

Comment:
I'm assuming the purpose of this whole block of code here is to keep timing consistent? As opposed to directly calling ->log() on each logger?

Reply:
yes, you check all the loggers and see which ones should be called, then you call them.  I thought this should be the dispatcher's job because it's supposed to be like a manager, and plus the dispatcher needs to wrap the log output of the loggers so that many logs can be included under one time stamp

Reply:
I was talking about why you're calling ->logThisLoop instead of ->log() directly here? It's getting a bit complex, so we can always discuss this in person if it's getting too complex to do here.

Reply:
The separation between checking and logging is because wrapStart and close should only be called when we're actually going to log something.  If we don't check beforehand then we won't know whether or not to open a new Json object (which is what the wrap is doing)

I guess logThisLoop could be set once we know that just one of the loggers should be called, and then we could check intervals in the second loop too, but either way we're iterating through the loggers twice so I figured one loop would be dedicated to evaluating intervals and the next to actual logging.

Reply:
Oh I see, so this actually groups together loggers with different intervals that are supposed to publish on the same second?

Reply:
^ yep!  That's how we get multiple logs under one time stamp when intervals overlap